packages/discord.js/src/managers/RoleManager.js

```diff
@@ -158,10 +169,17 @@ class RoleManager extends CachedManager {
       if (typeof icon !== 'string') icon = undefined;
     }
 
+    const colors = options.colors && {
+      primary_color: resolveColor(options.colors.primaryColor),
+      secondary_color: options.colors.secondaryColor && resolveColor(options.colors.secondaryColor),
+      tertiary_color: options.colors.tertiaryColor && resolveColor(options.colors.tertiaryColor),
+    };
+
     const data = await this.client.rest.post(Routes.guildRoles(this.guild.id), {
       body: {
         name,
         color,
+        colors,
```

Comment:
We shouldn't send color and colors here. Instead (if we keep the color option, which we shouldn't imho, see above) colors should be assigned the value `colors ??= {primary_color: color}` or rather as ternary when assigned.